Implement preview client

vauvenal5 opened the pull request:
Allows to request previews and thumbnails from the Nextcloud server. Closes #6 

I did not bump the version number since I do not know in what steps you are going. I guess this would be 1.2.0? I can do it if you want.

jld3103 commented:
Yes please bump the version to `1.2.0` and add a changelog and readme entry.

provokateurin commented on lib/src/preview/client.dart:
```diff
@@ -0,0 +1,62 @@
+
+  /// fetch preview for [remotePath] with provided [width] and [height]
+  /// [width] and [height] default to 512
+  Future<Uint8List> getPreview(String remotePath, {int width = 512, int height = 512}) async {
```
Are these default values also given in the java library? If not I would remove them so the user has to explicitly specify the dimensions.

vauvenal5 replied:
I removed the default values. Sizes are now mandatory.

provokateurin commented on lib/src/preview/client.dart:
```diff
@@ -0,0 +1,62 @@
+    final response = await _network.download('GET', _getThumbnailUrl(remotePath, width, height), [200]);
+    return response.stream;
+  }
+}
```
Missing new line

vauvenal5 replied:
Sorry not quite sure what you mean.

provokateurin replied:
Just add a final newline

vauvenal5 replied:
done

vauvenal5 commented:
I additionally added tests for preview, bumped the version and added to the changelog. I also renamed the test file to end in `_test.dart` according to the recommendations from the test package.

jld3103 commented:
Thanks for your contribution! Already published on https://pub.dev
